### PR TITLE
Range-ify std.string.chop

### PR DESCRIPTION
**Pull request by WalterBright**
Upgrade std.string.chop() to accept/return ranges.

100% unit test coverage.

http://dlang.org/phobos/std_string.html#.chop

### CODE REVIEW

File: std/string.d [hunk 1]
```diff
@@ -3059,19 +3062,67 @@ unittest
...
 
-    if (str.length >= 2 && str[$ - 1] == '\n' && str[$ - 2] == '\r')
-        return str[0 .. $ - 2];
-
-    str.popBack();
-
-    return str;
+    static if (isSomeString!Range)
+    {
+        if (str.length >= 2 && str[$ - 1] == '\n' && str[$ - 2] == '\r')
```
Comment (andralex) [inline std/string.d]: if (str.endsWith("\n\r"))

Comment (WalterBright) [inline std/string.d]: Don't want to do unnecessary autodecode, which endsWith will do.

Comment (andralex) [inline std/string.d]: oh yer right, I meant

    if (str.representation.endsWith("\n\r"))

Comment (WalterBright) [inline std/string.d]: "\r\n" will still autodecode

Comment (andralex) [inline std/string.d]: Tack `.representation` to that as well :D

**Comment by andralex** [comment]
Auto-merge toggled on

**Pull request merged**